src/context/NotificationProvider.tsx

```diff
@@ -0,0 +1,188 @@
+// src/context/NotificationProvider.tsx
+import React, {
+  createContext,
+  useEffect,
+  useCallback,
+  useContext,
+  PropsWithChildren,
+} from 'react';
+import {Alert} from 'react-native';
+import notifee, {
+  EventType,
+  Notification,
+  AndroidImportance,
+} from '@notifee/react-native';
+import messaging from '@react-native-firebase/messaging';
+
+import notificationService from '@/utils/notification.service';
+import {navigationRef} from './NavigationProvider';
+
+type NotificationContextType = {
+  goToNotificationSettings: () => void;
+  requestPermissions: () => Promise<boolean>;
+};
+
+type NotificationPayload = {
+  routeName: string;
+  screen?: string;
+  params?: string;
+};
+
+const NotificationContext = createContext<NotificationContextType>({
+  goToNotificationSettings: () => {},
+  requestPermissions: async () => false,
+});
+
+const NotificationProvider: React.FC<PropsWithChildren> = ({children}) => {
+  const onPressedNotification = useCallback(
+    async (notification: Notification) => {
+      if (!notification?.data) {
+        console.log('none data');
+        return;
+      }
+
+      try {
+        const payload = notification.data as NotificationPayload;
+        const parsedParams = payload.params
+          ? JSON.parse(payload.params)
+          : undefined;
+        if (payload.routeName && navigationRef.current?.isReady()) {
+          navigationRef.current.navigate(payload.routeName, {
+            screen: payload.screen,
+            params: parsedParams,
+          });
```

Comment:
```suggestion

        if (payload.routeName && navigationRef.current?.isReady()) {
           const parsedParams = payload.params
            ? JSON.parse(payload.params)
            : undefined;
        
          navigationRef.current.navigate(payload.routeName, {
            screen: payload.screen,
            params: parsedParams,
          });
```

if문 안에서만 parsedParams를 사용하니 조건문 통과후 JSON.parse를 진행해도 좋을 것 같네요